fix(db): ensure that flattened_errors is encoded as a JSON array

flrgh opened the pull request:
You know the deal: you create a list-like table but don't put anything in it, and all the sudden cjson thinks it's an object!

I wonder how many more times I'll fix this in my tenure at Kong :thinking:

Well, this one is fixed. I even added a little helper function for it.

KAG-6032

gszr reviewed (approved):
Nice catch. Can we add a regression test for it?

flrgh commented:
1. Added a test for the `cjson.encode(empty_table)`
2. Added a whole slew of unit tests by converting existing test cases in `spec/02-integration/04-admin_api/15-off_spec.lua`. This means we _sort of_ have some duplicate test coverage, but the unit tests are fast and will aid future devs in understanding how `declarative_config_flattened()` processes its input.
